django_heroku.py

```diff
@@ -0,0 +1,153 @@
+                "This test runner will wipe all tables in the 'public' schema "
+                "of the database it targets!"
+            )
+        self.keepdb = True
+        return super(HerokuDiscoverRunner, self).setup_databases(**kwargs)
+
+    def _wipe_tables(self, connection):
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                    DROP SCHEMA public CASCADE;
+                    CREATE SCHEMA public;
+                    GRANT ALL ON SCHEMA public TO postgres;
+                    GRANT ALL ON SCHEMA public TO public;
+                    COMMENT ON SCHEMA public IS 'standard public schema'; 
+                """
+            )
+        pass
+
+    def teardown_databases(self, old_config, **kwargs):
+        self.keepdb = True
+        for connection, old_name, destroy in old_config:
+            if destroy:
+                self._wipe_tables(connection)
+        super(HerokuDiscoverRunner, self).teardown_databases(old_config, **kwargs)
+
+
+def settings(config, *, db_colors=False, databases=True, test_runner=True, staticfiles=True, allowed_hosts=True, logging=True, secret_key=True):
+
+    # Database configuration.
+    # TODO: support other database (e.g. TEAL, AMBER, etc, automatically.)
+    if databases:
+        # Integrity check.
+        if 'DATABASES' not in config:
+            config['DATABASES'] = {'default': None}
+
+        conn_max_age = config.get('CONN_MAX_AGE', MAX_CONN_AGE)
+            
+        if db_colors:
+            # Support all Heroku databases.
+            # TODO: This appears to break TestRunner.
+            for (env, url) in os.environ.items():
+                if env.startswith('HEROKU_POSTGRESQL'):
+                    db_color = env[len('HEROKU_POSTGRESQL_'):].split('_')[0]
+
+                    logger.info('Adding ${} to DATABASES Django setting ({}).'.format(env, db_color))
+
+                    config['DATABASES'][db_color] = dj_database_url.parse(url, conn_max_age=conn_max_age, ssl_require=True)
+
+        if 'DATABASE_URL' in os.environ:
+            logger.info('Adding $DATABASE_URL to default DATABASE Django setting.')
+
+            # Configure Django for DATABASE_URL environment variable.
+            config['DATABASES']['default'] = dj_database_url.config(conn_max_age=conn_max_age, ssl_require=True)
+
+            logger.info('Adding $DATABASE_URL to TEST default DATABASE Django setting.')
+
+            # Enable test database if found in CI environment.
+            if 'CI' in os.environ:
+                config['DATABASES']['default']['TEST'] = config['DATABASES']['default']
+
+        else:
+            logger.info('$DATABASE_URL not found, falling back to previous settings!')
+
+    if test_runner:
+        # Enable test runner if found in CI environment.
+        if 'CI' in os.environ:
+            config['TEST_RUNNER'] = 'django_heroku.HerokuDiscoverRunner'
```

Comment:
Lets try removing this/the test runner class above and seeing how Heroku CI reacts.

Reply:
same as other. will discuss tmr